src/Libraries/RevitNodes/Elements/Dimension.cs

```diff
@@ -611,6 +652,44 @@ private static Autodesk.Revit.DB.XYZ GetMidpoint(Autodesk.Revit.DB.BoundingBoxXY
             return box.Min + ((box.Max - box.Min) / 2);
         }
 
+        /// <summary>
+        /// Whether the two vectors are perpendicular
+        /// </summary>
+        /// <param name="v1"></param>
+        /// <param name="v2"></param>
+        /// <returns></returns>
+        private static bool IsVertical(Autodesk.Revit.DB.XYZ v1, Autodesk.Revit.DB.XYZ v2)
+        {
+            var m = v1.X * v2.X + v1.Y * v2.Y + v1.Z * v2.Z;
+
+            if (m == 0)
+                return true;
+            else
+                return false;
+        }
+        private const double Tolerance = 0.000001;
```

Comment:
I think there maybe other place to have `Tolerance `, could you check about it? And also is there any existing library that has `IsVertical `or `IsParallel`.

Reply:
I just have a new commit, could you help review again?